department-of-veterans-affairs/vets-website
# [MHV-44568] Updated Navigation after sending a message

Summary

MHV-44568: Updated Navigation after User sends a Secure Message

User will be redirected to the original folder from where the message was sent from.
Originally the user was sent to the Inbox or Sent folder regardless of where the message was sent from.

MHV-44474: User Unable to Remove Custom Folder

When a user removes an EMPTY folder they will be presented with a modal to delete the folder.
When a user attempts to remove a folder with messages, they will be present with a modal that tells them to empty the folder before they can delete it.

Related issue(s)

https://jira.devops.va.gov/browse/MHV-44568
https://jira.devops.va.gov/browse/MHV-44474

Testing done
Unit tests added
QA testing completed

Screenshots
N/A

What areas of the site does it impact?

VA.GOV - Secure Messaging

Acceptance criteria

Quality Assurance & Testing

 I fixed|updated|added unit tests and integration tests for each feature (if applicable).
 No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
 Linting warnings have been addressed
 Documentation has been updated ([link to documentation](https://github.com/department-of-veterans-affairs/vets-website/compare/44568-2nd-User-is-able-to-navigate-after-sending-a-secure-message-mhv?expand=1#) *if necessary)
 Screenshot of the developed feature is added
 [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed
Error Handling

 Browser console contains no warnings or errors.
 Events are being sent to the appropriate logging solution
 Feature/bug has a monitor built into Datadog or Grafana (if applicable)
Authentication

 Did you login to a local build and verify all authenticated routes work as expected with a test user
⚠️ Team Sites (only applies to modifications made to the VA.gov header) ⚠️

 The vets-website header does not contain any web-components
 I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
 I reached out in the #sitewide-public-websites Slack channel for questions
Requested Feedback

(OPTIONAL) What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?
